ui/src/reusable_ui/components/inputs/ControlledInput.tsx

```diff
@@ -0,0 +1,177 @@
+// Libraries
+import React, {
+  Component,
+  CSSProperties,
+  ChangeEvent,
+  KeyboardEvent,
+} from 'react'
+import classnames from 'classnames'
+
+// Types
+import {ComponentStatus, ComponentSize, IconFont} from 'src/reusable_ui/types'
+
+// Styles
+import './Input.scss'
+
+export enum InputType {
+  Text = 'text',
+  Number = 'number',
+  Password = 'password',
+  Email = 'email',
+}
+
+interface Props {
+  value?: string
+  placeholder?: string
+  onChange?: (e: ChangeEvent<HTMLInputElement>) => void
+  onBlur?: () => void
+  onFocus?: () => void
+  onKeyPress?: (e: KeyboardEvent<HTMLInputElement>) => void
+  onKeyUp?: (e: KeyboardEvent<HTMLInputElement>) => void
+  onKeyDown?: (e: KeyboardEvent<HTMLInputElement>) => void
+  size?: ComponentSize
+  icon?: IconFont
+  status?: ComponentStatus
+  autoFocus?: boolean
+  spellCheck?: boolean
+  type?: InputType
+  widthPixels?: number
+  titleText?: string
+  disabledTitleText?: string
+  customClass?: string
+}
+
+class ControlledInput extends Component<Props> {
```

Comment:
The name is a bit confusing to me since it overloads the existing react terminology of “controlled” and “uncontrolled” inputs (e.g. https://reactjs.org/docs/forms.html#controlled-components).

Maybe it could be called `StatusInput`? `FormInput`? I dunno.

Reply:
I was thinking about building `ControlledTextArea` as well. My reasoning is that this component requires it to be controlled whereas using a standard `<input />` can be uncontrolled, though react throws a warning about doing so. It might be overkill and should just be called `Input`